src/wallet/interfaces.cpp

```diff
@@ -281,12 +281,12 @@ class WalletImpl : public Wallet
         CAmount& fee) override
     {
         LOCK(m_wallet->cs_wallet);
-        auto res = CreateTransaction(*m_wallet, recipients, change_pos,
+        auto res = CreateTransaction(*m_wallet, recipients, change_pos == -1 ? std::nullopt : std::make_optional(change_pos),
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/25273/commits/3d1a5f273c3c01b47ea4c74503c8353b2632dce8:

note for a follow-up: would nice if we could use this convention throughout the codebase instead of switching the magic number into a std::optional, like we are doing here.